doc/howto/security_harden.md

```diff
@@ -0,0 +1,275 @@
+(howto-security-harden)=
+# How to harden security for LXD
+
+To increase the security posture of your LXD deployment, review the following hardening recommendations and apply those relevant to your setup.
+
+## General
+
+(howto-security-harden-supported)=
+### Use a supported version
+
+Use only supported LTS releases or the latest feature release of LXD, and ensure that you update it regularly to receive security updates and bugfixes. See: {ref}`ref-releases`.
+
+(howto-security-harden-disable-unused)=
+### Disable unused features
+
+If you are not using certain LXD features, disable them to reduce the attack surface. For example:
+
+- Disable the UI:
+
+   ```bash
+   sudo snap set lxd ui.enable=false
+   sudo systemctl reload snap.lxd.daemon
+   ```
+
+- Delete unused networks and storage pools.
+
+## Access
+
+(howto-security-harden-group)=
+### Secure the `lxd` group
+
+Users in the `lxd` group who access LXD through the local Unix socket are given full administrative control over LXD. Thus, ensure that only trusted users are members of the `lxd` group (or any custom group you configure via `snap.lxd.daemon.group`). Audit group membership regularly.
+
+Also see: {ref}`howto-security-harden-restricted-group`.
+
+(howto-security-harden-remote)=
+### Harden remote API access
+
+For {ref}`authentication`, LXD can use either {abbr}`TLS (Transport Layer Security)` client certificates or OpenID Connect:
+
+- Client certificates:
+   - Ensure that only clients with certificates issued by your trusted Certificate Authority (CA) can connect. The {config:option}`server-core:core.trust_ca_certificates` option is `false` by default. To prevent auto-trusting of CA-signed certificates, ensure it remains disabled.
+   - Regularly audit and remove unused client certificates from the trust store.
+
+- OpenID Connect:
+   - Only set `oidc.client.secret` if required by the identity provider.
+   - Configure your OIDC provider to issue short-lived access tokens.
+   - Require multi-factor authentication (MFA) in your identity provider for accounts that can access LXD.
```

Comment:
I'm not familiar with how IdP work but I would not limit the recommendation for MFA to only accounts granting access to LXD. IMHO, everything you do with your IdP has the potential to be sensitive.

I'd assume that an IdP admin would be able to tweak which accounts can or cannot use LXD so that admin account needs to be equally protected by MFA.